crates/era-utils/src/export.rs

```diff
@@ -0,0 +1,260 @@
+//! Logic to export from database era1 block history
+//! and injecting them into era1 files with `Era1Writer`.
+
+use alloy_consensus::{BlockBody, BlockHeader, Header};
+use alloy_primitives::{BlockNumber, B256, U256};
+use eyre::{eyre, Result};
+use reth_era::{
+    era1_file::Era1Writer,
+    era1_types::{BlockIndex, Era1Id},
+    execution_types::{
+        Accumulator, BlockTuple, CompressedBody, CompressedHeader, CompressedReceipts,
+        TotalDifficulty,
+    },
+};
+use reth_fs_util as fs;
+use reth_storage_api::{BlockNumReader, BlockReader, HeaderProvider};
+use std::{
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+use tracing::{info, warn};
+
+const REPORT_INTERVAL_SECS: u64 = 10;
+const ENTRY_HEADER_SIZE: usize = 8;
+const VERSION_ENTRY_SIZE: usize = ENTRY_HEADER_SIZE;
+
+/// Configuration to export block history
+/// to era1 files
+#[derive(Clone, Debug, Default)]
+pub struct ExportConfig {
+    /// Directory to export era1 files to
+    pub dir: PathBuf,
+    /// First block to export
+    pub first_block_number: BlockNumber,
+    /// Last block to export
+    pub last_block_number: BlockNumber,
+    /// Number of blocks per era1 file
+    pub step: u64,
```

Comment:
The amount of blocks per ERA1 file is capped at 8192: https://github.com/eth-clients/e2store-format-specs/blob/main/formats/era1.md

The importer assumes that every file contains 8192 blocks except the last one.

I think the `step` should not be configurable and simply set to that constant.

```suggestion
```

Reply:
can't hurt if we have a setting for this and use the epoch consts as default imo

maybe there's some usecase for dumping them in one big file